src/OpenDiffix.Core/QueryEngine.fs

```diff
@@ -15,6 +15,32 @@ let rec private extractColumns query =
 
 type QueryResult = { Columns: Column list; Rows: Row list }
 
+// The main entrypoint to the `OpenDiffix.Core` library.
+// Takes in a SQL `statement` as main parameter along with a set of
+// parameters in `queryContext`.
+//
+// Returns the query result as a list of data rows, along with the
+// columns specifications.
+//
+// Supports two main modes of operation:
+// I. Standard SQL statement, with `aidColumns` specified within the `queryContext`
+//
+// Will rewrite the statement so that the query result is anonymized according to the
+// `aidColumns` provided. For example, it will append `AND NOT diffix_low_count(aidColumns)`
+// to the `HAVING` SQL clause and rewrite regural SQL aggregators to Diffix aggregators
+// (e.g. `count` -> `diffix_low_count`, see below)
+//
+// II. SQL statement with Diffix aggregators, no `aidColumns` specified within the `queryContext`
+//
+// (**NOTE**) some AID columns may (and should) be used within the Diffix aggregator functions
+//
+// Will not rewrite the statement. The caller is expected to use Diffix aggregator functions
+// manually, in order to anonymize the query result manually.
```

Comment:
Doesn't add too much information

Reply:
which part exactly? The two operation modes where very confusing to me, when I was first confronted with this. This distinction is important to make in my opinion, to capture the somewhat "leaky" abstraction, which is OK to have, but confusing, if not made explicit.

My aim is to make the answer to the question "which of the components is responsible for the anonymization?" hard to miss

Reply:
I see what you mean. We used to call these standard queries and anonymizing queries. OK with you if I tweak the comments a bit to match with our terminology?

Reply:
hm, calling these `standard` vs `anonymizing` queries doesn't fully make it clear. I want to document the intended use of this function (or the entire lib, since this is its entrypoint) to anyone without relying to terminology coined elsewhere, ELI5 style. But I'm working on a significant revamp, I hope it will read better then